### PR TITLE
added getElementalGradient function

### PR DESCRIPTION
**Pull request by e-dub**
**📝 Description**
Quick fix by adding GetElementalGradient function analogously to GetNodalGradient to allow usage like:
`stressNablaE = response_function.GetElementalGradient(sm.YOUNG_MODULUS_SENSITIVITY)`

**🆕 Changelog**
- Added function GetElementalGradient to structural_response.py

### CODE REVIEW

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 1]
```diff
@@ -295,6 +323,21 @@ def GetNodalGradient(self, variable):
...
+                            StructuralMechanicsApplication.YOUNG_MODULUS_SENSITIVITY,
+                            StructuralMechanicsApplication.THICKNESS_SENSITIVITY,
+                            StructuralMechanicsApplication.I22_SENSITIVITY,
+                            StructuralMechanicsApplication.I33_SENSITIVITY,
+                            ]:
+            raise RuntimeError("GetElementalGradient: No gradient for {}!".format(variable.Name))
+        gradient = {}
+        for element in self.adjoint_model_part.Elements:
+            gradient[element.Id] = element.GetValue(variable)
+            #gradient[element.Id] = element.GetSolutionStepValue(variable)
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py]:
```suggestion
```

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 2]
```diff
@@ -295,6 +323,21 @@ def GetNodalGradient(self, variable):
             gradient[node.Id] = node.GetSolutionStepValue(variable)
         return gradient
 
+    def GetElementalGradient(self, variable):
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py:305]: I think this method should also be added to the `ResponseFunctionInterface` with a `NotImplementedError`

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 3]
```diff
@@ -135,6 +149,8 @@ def __init__(self, identifier, response_settings, model):
 
         self.primal_analysis = StructuralMechanicsAnalysis(model, ProjectParametersPrimal)
         self.primal_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.SHAPE_SENSITIVITY)
+        self.primal_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.YOUNGS_MODULUS_SENSITIVITY)
+
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py]: I think this EigenFrequencyResponseFunction,  StrainEnergyResponseFunction (and MassResponsefunction?) are only implemented for shape sensitivities at the moment.

Note: there is a difference in the implementation of these first response functions and the "AdjointResponseFunction" below, that has a more general implementation that also allows element properties to be used as variables.

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 4]
```diff
@@ -92,6 +92,20 @@ def GetNodalGradient(self, variable):
             gradient[node.Id] = node.GetSolutionStepValue(variable)
         return gradient
 
+    def GetElementalGradient(self, variable):
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py:95]: see comment below, i think this is not implemented.

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py (hunk 3 above)
Comment (e-dub) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py]: I can confirm that these routines return dm/dE, dm/dt, dU/dE, dU/dt = 0. Removed and added error.

Comment (e-dub) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py]: well obviously a couple should be zero, but those that are obvioiusly neq 0 are zero.

Comment (armingeiser) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py]: Explanation: `EigenFrequencyResponseFunction, StrainEnergyResponseFunction, MassResponsefunction` have been the first implementations of response functions in Kratos, specifically with application in ShapeOptimization in mind. 
All other response functions with "adjoint" in there name, have been implemented by @MFusseder following a much more general structure that allows computing gradients w.r.t different variables than shape only.

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py (hunk 2 above)
Comment (armingeiser) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py:305]: i just see its already there.

**Review by armingeiser — APPROVED** [review]

**Pull request merged**